bundles/core/src/main/java/com/adobe/cq/wcm/core/components/commons/link/LinkHandler.java

```diff
@@ -0,0 +1,55 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2022 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.commons.link;
+
+import org.apache.sling.api.resource.Resource;
+import org.jetbrains.annotations.NotNull;
+
+import com.day.cq.wcm.api.Page;
+
+/**
+ * Computes a link based on input configuration.
+ */
+public interface LinkHandler {
+
+    /**
+     * Returns a link builder based on the given resource.
+     *
+     * @param resource Resource to read the link properties from.
+     * @return {@link LinkBuilder}
+     */
+    @NotNull
+    LinkBuilder get(@NotNull Resource resource);
```

Comment:
This is always a new(!) LinkBuilder, therefore `create()` would be a better fit than `get()`, but in this case I would just add a constructor to LinkBuilder (more obvious how to create) and get rid of this interface all together!

Reply:
I also prefer `get` as in essence it is returning a link based on properties defined by the Resource.

Reply:
> it's not unlikely that the link builder impl will need to be injected some references or other stuff, so reducing this to a constructur of linkbuilder will make that difficult.

Ok, then what about renaming this class to `LinkBuilderFactory`?

Reply:
the goal is not to create `LinkBuilder` instances (this is only an intermediate step), but to create `Link` instances. in that direction `LinkFactory` could be ok. (on the other hand having two classes `LinkFactory` and `LinkBuilder` in the API is also a bit confusing for the user)

Reply:
The API requires two steps:
First creating the link builder and then creating the link out of that (that is the case right now).
IMHO it would be clearer to express the different roles in the class name, therefore `LinkBuilderFactory` (to create the `LinkBuilder`) and then `LinkBuilder` (to create the actual link) is reasonable for me.
OTOH `LinkBuilder` and `LinkFactory` would be highly confusing as the latter would not create links but rather builder instances.

Reply:
i agree that `LinkFactory` is not a good name for this reasons.

in general i'm fine with both `LinkHandler` and `LinkBuilderFactory`. i've a slight preference to the former as it might be we need additional Link-related methods in the link handler in the future which do not return a LinkBuilder. for example, in https://wcm.io/handler/link/apidocs/io/wcm/handler/link/LinkHandler.html we have a method "invalid" which returns an invalid links. which looks odd, but as a link object is never null it's otherwise difficult if a component has to return an invalid link on purpose e.g. if no page, resource or string as input is available. maybe that's an edge case which could be solved otherwise.

Reply:
For me `LinkHandler` is kind of fuzzy and not really explaining well the purpose of this class, therefore I have a strong preference towards `LinkBuilderFactory` :-)

Reply:
As @kwin noted, this proposal imposes a 2-steps definition for a link. We could simplify it by getting rid of the `LinkBuilder` and only introduce a `LinkHandler` interface as follows (if this idea gets traction I would create a 4th draft with detailed javadocs):
```
    Link getLink(@NotNull Resource resource);
    Link getLink(@NotNull Resource resource, @Nullable String linkURLPropertyName);
    Link getLink(@NotNull Page targetPage);
    Link getLink(@NotNull String URL);
    void setLinkTarget(Link link, @Nullable String linkTarget);
    void setLinkAttribute(Link link, String key, String value);
```
WDYT?

Reply:
any other ideas to avoid the 2 steps to get a link?

Reply:
+1 for `LinkManager`

Reply:
LinkManager is fine from my side as well